### PR TITLE
change cvn validation logic

### PR DESCRIPTION
**Pull request by hakiemaul**
### Problem
We are not validating CVN correctly based on card type. Will cause an error when creating charge, better to handle it before creating the token

### Solution
Change the CVN validation logic to check card type

### CODE REVIEW

File: xendit-android/src/main/java/com/xendit/utils/CardValidator.java [hunk 1]
```diff
@@ -91,6 +91,30 @@ public static boolean isExpiryValid(String expirationMonth, String expirationYea
...
+    public static boolean isCvnValid(String cardCVN, String cardNumber) {
+        if (cardCVN == null || cardNumber == null) {
+            return false;
+        }
+
+        String cleanCvn = cleanCvn(cardCVN);
+        String cleanCardNumber = cleanCardNumber(cardNumber);
+
+        if (isNumeric(cleanCvn) && Integer.parseInt(cleanCvn) >= 0) {
+            if ((isCardAmex(cleanCardNumber) && cleanCvn.length() == 4) || (!isCardAmex(cleanCardNumber) && cleanCvn.length() == 3)) {
```
Comment (boxbag) [inline xendit-android/src/main/java/com/xendit/utils/CardValidator.java]: just return the expression

File: xendit-android/src/main/java/com/xendit/utils/CardValidator.java [hunk 2]
```diff
@@ -91,6 +91,26 @@ public static boolean isExpiryValid(String expirationMonth, String expirationYea
...
+    public static boolean isCvnValid(String cardCVN, String cardNumber) {
+        if (cardCVN == null || cardNumber == null) {
+            return false;
+        }
+
+        String cleanCvn = cleanCvn(cardCVN);
+        String cleanCardNumber = cleanCardNumber(cardNumber);
+
+        if (isNumeric(cleanCvn) && Integer.parseInt(cleanCvn) >= 0) {
+            return ((isCardAmex(cleanCardNumber) && cleanCvn.length() == 4) || (!isCardAmex(cleanCardNumber) && cleanCvn.length() == 3));
```
[the remark this answers is not in the document]
Comment (0505gonzalez) [reply xendit-android/src/main/java/com/xendit/utils/CardValidator.java]: `return isCardAmex(cleanCardNumber) ? cleanCvn.length() == 4 : cleanCvn.length() == 3`

File: xendit-android/src/main/java/com/xendit/Xendit.java [hunk 3]
```diff
@@ -162,6 +162,11 @@ private void createSingleOrMultipleUseToken(final Card card, final String amount
                 return;
             }
 
+            if (card.getCreditCardCVN() != null && !CardValidator.isCvnValidForCardType(card.getCreditCardCVN(), card.getCreditCardNumber())) {
```
Comment (0505gonzalez) [inline xendit-android/src/main/java/com/xendit/Xendit.java]: `card.getCreditCardCVN() != null` i think is redundant. doesnt `isCvnValidForCardType` hanldle it already?

File: xendit-android/src/main/java/com/xendit/utils/CardValidator.java [hunk 4]
```diff
@@ -86,6 +86,28 @@ public static boolean isExpiryValid(String expirationMonth, String expirationYea
     }
 
     /**
+     * Determines whether the card CVN length is valid
+     *
+     * @param  cardCVN The credit card CVN
+     * @param  cardNumber The credit card number
+     * @return true if the cvn length is valid for this card type, false otherwise
+     */
+    public static boolean isCvnValidForCardType (String cardCVN, String cardNumber) {
```
Comment (0505gonzalez) [inline xendit-android/src/main/java/com/xendit/utils/CardValidator.java]: Forgot to mention, this should have unit test coverage

File: xendit-android/src/main/java/com/xendit/Xendit.java (hunk 3 above)
Comment (hakiemaul) [reply xendit-android/src/main/java/com/xendit/Xendit.java]: this is the same like iOS, we don't want to validate if there is no CVN input

**Comment by 0505gonzalez** [comment]
LGTM

**Comment by hakiemaul** [comment]
LGTM

**Pull request merged**